contexts/ModalContext.tsx

```diff
@@ -26,15 +27,32 @@ const ModalContext = createContext<Value | null>(null);
 export const ModalContextProvider: FC<PropsWithChildren> = ({ children }) => {
   const [current, setCurrent] = useState<Identifier>();
   const [args, setArgs] = useState<Args<Identifier>>();
-  const open = useCallback((identifier: Identifier, _args: Args<Identifier>) => {
-    setArgs(_args);
-    setCurrent(identifier);
-  }, []);
+  const router = useRouter();
+  const open = useCallback(
+    (identifier: Identifier, _args: Args<Identifier>) => {
+      setArgs(_args);
+      setCurrent(identifier);
+      router.replace({
+        query: { ...router.query, modal: identifier },
+      });
+    },
+    [router],
+  );
 
   const close = useCallback(() => {
     setArgs(undefined);
     setCurrent(undefined);
-  }, []);
+    const { query } = router;
+    delete query.modal;
+    router.replace(query);
```

Comment:
Prefer not to mutate the object
```suggestion
    const { query } = router;
    const { modal, ...rest } = router;
    router.replace(rest)
```